core/struct.rbs

```diff
@@ -1,299 +1,51 @@
-#
-# The member values may be managed thus:
-#
-#     joe.name    # => "Joe Smith"
-#     joe.name = 'Joseph Smith'
-#     joe.name    # => "Joseph Smith"
-#
-# And thus; note that member name may be expressed as either a string or a
-# symbol:
-#
-#     joe[:name]  # => "Joseph Smith"
-#     joe[:name] = 'Joseph Smith, Jr.'
-#     joe['name'] # => "Joseph Smith, Jr."
-#
-# See Struct::new.
-#
-# ## What's Here
-#
-# First, what's elsewhere. Class Struct:
-#
-# *   Inherits from [class Object](rdoc-ref:Object@What-27s+Here).
-# *   Includes [module Enumerable](rdoc-ref:Enumerable@What-27s+Here), which
-#     provides dozens of additional methods.
-#
-#
-# See also Data, which is a somewhat similar, but stricter concept for defining
-# immutable value objects.
-#
-# Here, class Struct provides methods that are useful for:
-#
-# *   [Creating a Struct
-#     Subclass](rdoc-ref:Struct@Methods+for+Creating+a+Struct+Subclass)
-# *   [Querying](rdoc-ref:Struct@Methods+for+Querying)
-# *   [Comparing](rdoc-ref:Struct@Methods+for+Comparing)
-# *   [Fetching](rdoc-ref:Struct@Methods+for+Fetching)
-# *   [Assigning](rdoc-ref:Struct@Methods+for+Assigning)
-# *   [Iterating](rdoc-ref:Struct@Methods+for+Iterating)
-# *   [Converting](rdoc-ref:Struct@Methods+for+Converting)
-#
-#
-# ### Methods for Creating a Struct Subclass
-#
-# *   ::new: Returns a new subclass of Struct.
-#
-#
-# ### Methods for Querying
-#
-# *   #hash: Returns the integer hash code.
-# *   #length, #size: Returns the number of members.
-#
-#
-# ### Methods for Comparing
-#
-# *   #==: Returns whether a given object is equal to `self`, using `==` to
-#     compare member values.
-# *   #eql?: Returns whether a given object is equal to `self`, using `eql?` to
-#     compare member values.
-#
-#
-# ### Methods for Fetching
-#
-# *   #[]: Returns the value associated with a given member name.
-# *   #to_a, #values, #deconstruct: Returns the member values in `self` as an
-#     array.
-# *   #deconstruct_keys: Returns a hash of the name/value pairs for given member
-#     names.
-# *   #dig: Returns the object in nested objects that is specified by a given
-#     member name and additional arguments.
-# *   #members: Returns an array of the member names.
-# *   #select, #filter: Returns an array of member values from `self`, as
-#     selected by the given block.
-# *   #values_at: Returns an array containing values for given member names.
-#
-#
-# ### Methods for Assigning
-#
-# *   #[]=: Assigns a given value to a given member name.
-#
-#
-# ### Methods for Iterating
-#
-# *   #each: Calls a given block with each member name.
-# *   #each_pair: Calls a given block with each member name/value pair.
-#
-#
-# ### Methods for Converting
-#
-# *   #inspect, #to_s: Returns a string representation of `self`.
-# *   #to_h: Returns a hash of the member name/value pairs in `self`.
-#
-class Struct[Elem] < Object
+class Struct[Elem]
   include Enumerable[Elem?]
```

Comment:
Why `Elem?` with `?` in so many places? If the user would like to have, say, String or `nil` for values, can’t they have a `Struct[String?]`?

Reply:
It's cause there's no way to guarantee every element is assigned. Since you can do `My3dPoint.new(1)`, the extra elements are assigned as nil. Thus, anything that interacts with them are `Elem?`.